routes/api/auth.js

```diff
@@ -1,10 +1,88 @@
 //Handle getting a JSON webtoken for authorization.
 const express = require('express');
 const router = express.Router();
+const bcrypt = require('bcryptjs');
+const auth = require('../../middleware/auth');
+const jwt = require('jsonwebtoken');
+const config = require('config');
+const { check, validationResult } = require('express-validator/check');
+
+//Bringing in the User model.
+const User = require('../../models/User');
 
 // @route   GET api/auth
 // @desc    Test route
 // @access  Public
-router.get('/', (req, res) => res.send('Auth route'));
+
+//The route should return the user's data.
+router.get('/', auth, async (req, res) => {
+  try {
+    //This is a protected route. We can access req.user anywhere from a protected route.
+    const user = await User.findById(req.user.id).select('-password');
+    res.json(user);
+  } catch (err) {
+    console.log(err.message);
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   POST api/auth
+// @desc    Authenticate user & get token
+// @access  Public (Purpose is to get the token so you can make requests to private routes)
+router.post(
+  '/',
+  [
+    check('email', 'Please include a valid email').isEmail(),
+    check('password', 'Password is required').exists(),
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      //If they don't include above information correctly, then this a bad request.
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    const { email, password } = req.body;
+
+    try {
+      //Check if the user exits
+      let user = await User.findOne({ email });
```

Comment:
One can use projection here to only retrieve the necessary fields. See documentation:
https://mongoosejs.com/docs/api.html#model_Model.findOne
https://mongoosejs.com/docs/api.html#query_Query-select

I realize I didn't define query when we were speaking earlier.

Query: a request (usually a find() or read request) made to the database. findOne({conditions}) is a query with conditions.